stdlib/buffer.mli

```diff
@@ -30,6 +30,19 @@
 
 *)
 
+(** {b Unsynchronized accesses} *)
+
+[@@@alert unsafe_unsynchronized_access
+    "Unsynchronized accesses to buffers are a programming error."
+]
+
+ (**
+    Unsynchronized accesses might break module abstraction and create invalid
```

Comment:
"Breaking module abstraction" sounds quite abstract. Is that needed?

Can we perhaps go with 

> Unsynchronized access to a buffer may lead to an invalid buffer state. Thus, concurrent uses of a buffer must be synchronized (for instance with a {!Mutex.t}).

I'm also avoiding the term "sequentialized" as it is unclear what that means.

Reply:
Creating an invalid state is a better phrasing indeed. I would rather avoid switching between `access` and `uses` between the two sentence. It is also not clear to me if there a case where a single unsynchronized access can create an invalid buffer state, thus `accesses` seems better to me:

>Unsynchronized accesses to a buffer may lead to an invalid buffer state. Thus, concurrent accesses to a buffer must be synchronized (for instance with a {!Mutex.t}).

?

Reply:
LGTM.